### ISSUE TITLE
Make a plan for filename length restrictions

### ISSUE THREAD

**Issue by julianbrost** [Icinga/icinga2#8844]
Icinga uses the object name for naming files for API-created objects. There were and still are issues due to object names being longer than what file systems allow (typically max 255 bytes for everything Icinga should be running on), see for example #8022.

There were efforts to improve this (#8025, #8835), however with some restrictions on object types to minimize issues with objects not being synced properly throughout the cluster. The fundamental issue is that a fix has to be available on lower levels of the cluster first, however these are typically (as that's the recommended and only supported way) updated last.

So we need to make a plan how to get a fixed version there while not having sync issues all the time in the meantime, maybe something like I suggested in https://github.com/Icinga/icinga2/pull/8835#discussion_r656952157.

**Comment by Al2Klimov** [comment]
## Blocked by

* #8834

(Isn't it?)

**Comment by julianbrost** [comment]
#8834 for sure can be a part of the solution. It's just that the overall issue might need some coordinated changes over 3 major releases, so the intention of this issue is to discuss where we want to be in the end and how to get there.

**Comment by Al2Klimov** [comment]
Do you agree that

1. v2.14 must have #8834
2. In v2.16 we could wildly change new files' paths if we want

?

**Comment by julianbrost** [comment]
Yes, we should do that with the next release, otherwise it would just delay our options.
